### PR TITLE
Unfurling of Slack links

### PR DESCRIPTION
**Pull request by tommoor**
- If there is not an authentication in the db for slack then we prompt them to install the app.
- Any documents that the user tries to unfurl but doesn't have access to will be silently skipped.
- Otherwise a successful unfurl looks something like:

image

(the color comes from the collection color 😄 )

### CODE REVIEW

File: app/scenes/Settings/Slack.js [hunk 1]
```diff
@@ -17,11 +17,11 @@ class Slack extends Component {
         <h1>Slack</h1>
         <HelpText>
           Connect Outline to your Slack team to instantly search for documents
-          using the <Code>/outline</Code> command.
+          using the <Code>/outline</Code> command and unfurl Outline links.
```
Comment (jorilallo) [inline app/scenes/Settings/Slack.js]: I think "preview Outline links" would be more descriptive than "unfurl"

File: server/api/hooks.js [hunk 2]
```diff
@@ -1,10 +1,56 @@
...
+  let unfurls = {};
+  for (let link of event.links) {
+    const id = link.url.substr(link.url.lastIndexOf('/') + 1);
+    const doc = await Document.findById(id);
+    if (!doc || doc.teamId !== user.teamId) continue;
+
+    unfurls[link.url] = {
+      title: doc.title,
+      text: doc.getSummary(),
+      color: doc.collection.color || undefined,
```
Comment (jorilallo) [inline server/api/hooks.js]: Is there a difference between null and undefined here?

File: server/models/Event.js [hunk 3]
```diff
@@ -9,30 +9,6 @@ const Event = sequelize.define('event', {
   },
   name: DataTypes.STRING,
   data: DataTypes.JSONB,
-
```
Comment (jorilallo) [inline server/models/Event.js]: ?

Comment (tommoor) [reply server/models/Event.js]: You don't need to define these, the `belongsTo` call takes care of it

File: server/api/hooks.js (hunk 2 above)
Comment (tommoor) [reply server/api/hooks.js]: Not sure, probably not - was just being cautious

File: app/scenes/Settings/Slack.js (hunk 1 above)
Comment (tommoor) [reply app/scenes/Settings/Slack.js]: agree

**Comment by tommoor** [comment]
I've added `SLACK_APP_ID` added to production env

File: .env.sample [hunk 4]
```diff
@@ -7,19 +9,26 @@ DATABASE_URL_TEST=postgres://user:pass@localhost:5432/outline-test
...
 PORT=3000
 REDIS_URL=redis://redis:6379
-SLACK_KEY=71315967491.XXXXXXXXXX
-SLACK_SECRET=d2dc414f9953226bad0a356c794XXXXX
 URL=http://localhost:3000
 DEPLOYMENT=hosted
 ENABLE_UPDATES=true
+
+# Third party credentials (optional)
+SLACK_KEY=71315967491.XXXXXXXXXX
```
Comment (jorilallo) [inline .env.sample]: KEY and SECRET are actually mandatory right now for auth, but all others optional, right?

**Pull request merged**